tests/shardddl3/run.sh

```diff
@@ -871,6 +871,1199 @@ function DM_112() {
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a,b));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a,b));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b int, primary key(a,b));\"" \
+        "clean_table" "optimistic"
+}
+
+function DM_134_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} drop primary key, add primary key(b);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb1} drop primary key, add primary key(b);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb2} drop primary key, add primary key(b);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(10,10);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(11,11);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(12,12);"
+
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+# Change the primary key field.
+function DM_134 {
+    run_case 134 "double-source-pessimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b int, primary key(a));\"" \
+        "clean_table" "pessimistic"
+    run_case 134 "double-source-optimistic" \
+        "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb1} (a int, b int, primary key(a));\"; \
+         run_sql_source2 \"create table ${shardddl1}.${tb2} (a int, b int, primary key(a));\"" \
+        "clean_table" "optimistic"
+}
+
+function DM_135_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3);"
+
+    if ! run_sql_source1 "alter table ${shardddl1}.${tb1} drop primary key, add primary key(b);" 2>&1 | \
+        grep "Incorrect table definition; there can be only one auto column and it must be defined as a key" >/dev/null;
+    then
+        echo "sql should be failed because there can be only one auto column and it must be defined as a key" >&2
+        return 255
+    fi
```

Comment:
Do we need to test the behavior of MySQL? 🤔

Reply:
The error here is directly reported by MySQL. MySQL doesn't allow such a behavior.